[CORL-1121] Remove username validation when updated by admin

nick-funk opened the pull request:
## What does this PR do?

Applies no validation when an admin updates a username.

## What changes to the GraphQL/Database Schema does this PR introduce?

None

## How do I test this PR?

Use the following mutation:

```
mutation ($id: ID!, $name: String!, $clientMutationId: String!) {
  updateUserUsername(
    input: {
      userID: $id,
      username: $name,
      clientMutationId: $clientMutationId
    })
      {
        user {
          id
          username
        }
      }
}
```

where:
- `$id` is some valid userID.
- `$name` is a username with spaces in it.
- `$clientMutationId` is `"1"`

nick-funk committed:
Use SSO validation for updateUserUsername

CORL-1121

nick-funk commented on src/core/server/services/users/users.ts:
```diff
@@ -631,7 +632,7 @@ export async function updateUsernameByID(
 ) {
   // Validate the username.
-  validateUsername(username);
+  validateUsername(username, SSO_USERNAME_REGEX);
```
This is the update call used by the `updateUserUsername` mutator.

@wyattjoh is this sufficient? Or should we load up the user and confirm they are an SSO account before passing this different RegEx?

If we do want to load the user and confirm they are SSO, what fields should we be using to determine the user is an SSO user?

wyattjoh replied:
For this mutation, I'd just remove the username validation here all together. It can only be used by the admin's, and if they want to change it or allow it through their interfaces, we can't stop it.

```suggestion

```

nick-funk committed:
Remove validation on admin-only user rename

We don't need to validate the user-name as
only admin's will be using this mutator. If
they're updating things incorrectly, that's
up to them to manage with this flow.

CORL-1121
